connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -213,7 +213,9 @@ class SparkSession private[sql] (
    * @param sqlText
    *   A SQL statement with named parameters to execute.
    * @param args
-   *   A map of parameter names to literal values.
+   *   A map of parameter names to string values that are parsed as SQL literal expressions.
```

Comment:
```suggestion
   *   A map of parameter names to string values that are parsed as fragments of SQL text and interpreted as literal expressions. The parameter bindings do not provide any
   safety guarantees on how the SQL fragment is interpreted and should not be treated
   as fixed values like in prepared statements.
```

I think we need to be much much clearer on what the actual behavior is.

Reply:
> that are parsed as fragments of SQL text and interpreted as literal expressions.

We used `parseExpression()` to parse the input string values to single expression. That the comment describes directly. We don't interpret anything.

Reply:
> should not be treated as fixed values like in prepared statements.

I didn't get why it should be treated in this way. IMHO, this statement is useless.

Reply:
```
DATE'2023-03-21'
```
Essentially says parse this literal as a date and the output expression will be a date. This means you're interpreting the input, am I mistaken?

The core part is that users of the API will see that it supports parameter substitution and will believe it provides the same guarantees of fixed literals as for example prepared statements which is not the case. You're not able to perform full SQL injection using this API but it's enough that you can't blindly mix trusted and untrusted input.

Reply:
> This means you're interpreting the input, am I mistaken?

For the "interpreting" word means execute the SQL text and produce a date, but we just try to convert the input text to a literal. Roughly speaking, switching by branches to parse the input. Don't think that it can be considered as interpretation (or evaluation, execution and so on).

> The core part is that users of the API will see that it supports parameter substitution and will believe it provides the same guarantees of fixed literals ...

It provides, we checks that all args are literals but arbitrary expressions, see https://github.com/apache/spark/blob/f8966e7eee1d7f2db7b376d557d5ff6658c80653/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/parameters.scala#L92

Reply:
What I say remains true, you have to have significant internal knowledge of how the implementation works to understand what happens. For the user of the API, it's not clear how the input values are (parsed|interpreted|processed).

This is a security concern and should be treated as such.

Reply:
> For the user of the API, it's not clear how the input values are (parsed|interpreted|processed).

I think the SQL standard defines the syntax of literal for each data type, which doesn't require internal knowledge.